packages/jest-haste-map/src/crawlers/watchman.js

```diff
@@ -51,48 +51,52 @@ module.exports = async function watchmanCrawl(
 
   try {
     const watchmanRoots = new Map();
-    for (const root of roots) {
-      const response = await cmd('watch-project', root);
-      const existing = watchmanRoots.get(response.watch);
-      // A root can only be filtered if it was never seen with a relative_path before
-      const canBeFiltered = !existing || existing.length > 0;
+    await Promise.all(
+      roots.map(async root => {
+        const response = await cmd('watch-project', root);
+        const existing = watchmanRoots.get(response.watch);
+        // A root can only be filtered if it was never seen with a relative_path before
+        const canBeFiltered = !existing || existing.length > 0;
 
-      if (canBeFiltered) {
-        if (response.relative_path) {
-          watchmanRoots.set(
-            response.watch,
-            (existing || []).concat(response.relative_path),
-          );
-        } else {
-          // Make the filter directories an empty array to signal that this root
-          // was already seen and needs to be watched for all files/directories
-          watchmanRoots.set(response.watch, []);
+        if (canBeFiltered) {
+          if (response.relative_path) {
+            watchmanRoots.set(
+              response.watch,
+              (existing || []).concat(response.relative_path),
+            );
+          } else {
+            // Make the filter directories an empty array to signal that this root
+            // was already seen and needs to be watched for all files/directories
+            watchmanRoots.set(response.watch, []);
```

Comment:
Slightly concerned about race conditions here but it should be fine since the main thread is memory-safe, right?

Reply:
yes, this is completely fine